### PR TITLE
Ports Automatic Station Stabilizer Bots from Bee (ASSBot)

### PR DESCRIPTION
**Pull request by maxion12345**
PORT: https://github.com/BeeStation/BeeStation-Hornet/pull/2432
BUT I CHANGED IT

# General Documentation

### Intent of your Pull Request

Adds the AtmosBot! A robot designed to BLOW ITSELF UP OVER ANY HOLES IN THE FLOOR

BEHOLD MY CODERSPRITE

Simple recipe: robot arm, smartmetal grenade, prox sensor, analyzer

When emagged, it'll fucking explode instead of spewing foam

### Why is this change good for the game?
Variety is the spice of life. Also, suicide foam bot work better
# Wiki Documentation
The recipe is cyborg arm onto analyzer, then grenade onto that assembly, then prox sensor onto that.
### Briefly describe your PR and the impacts of it, in layman's terms. 
Roboticists will need an updated wiki page to reflect the new robot

### What should players be aware of when it comes to the changes your PR is implementing?
see above
### What general grouping does this PR fall under? 
add

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
Each bot can only deploy (1) holofan, and will stay near the holofan until the breach is repaired.

# Changelog

Edit this changelog for changes that are noticeable by the players. Remove it if this isn't the case. If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name. Prefix the PR title with [admin] if it's something admin related. Prefix the PR title with [s] if you are fixing an exploit so it is not announced on discord and the server.

:cl:  PowerfulBacon & Wuba
rscadd: Ported the Atmosbot from Beestation! Modified for Yogurt station, these robots are foam grenades with legs and will seal breaches or die trying
/:cl:

### CODE REVIEW

**Comment by Cartlord** [comment]
These seem quite ridiculously strong for Yogs, IMO; I think they fit Bee more, since over there it's a genuine possibility that nobody knows how to un-fuck Atmos - but here, we usually have at least someone able to fix a breach.
EDIT: Also, what's to stop Cargo Techs from using an EMagged Atmosbot in Yet Another Miasma Farm Setup? Might wanna replace that with a deadlier gas (BZ maybe? Though that could also be abused by Atmos) rather than a mild inconvenience.

**Comment by maxion12345** [comment]
> These seem quite ridiculously strong for Yogs, IMO; I think they fit Bee more, since over there it's a genuine possibility that nobody knows how to un-fuck Atmos - but here, we usually have at least someone able to fix a breach.
> EDIT: Also, what's to stop Cargo Techs from using an EMagged Atmosbot in Yet Another Miasma Farm Setup? Might wanna replace that with a deadlier gas (BZ maybe? Though that could also be abused by Atmos) rather than a mild inconvenience.

Nah, they aren't super strong. Remember, each bot can only deploy one holofan so you need at minimum four to seal a 1x1 breach. And while sealing breaches they'll be unable to refill rooms. The miasma farm emagged bot is an interesting idea, but gas exports were nerfed after everyone in their mother did one. In addition, emags are contraband and technically weapons under the new space law, so if security hands one over to cargo then admins gonna get salty.

**Comment by stwordcitizen** [comment]
Less work that atmos can do, therefor less reason to play the role if your not planning on doing autism projects in atmos.

**Comment by UselessTheremin** [comment]
these sprites are really gross

**Comment by maxion12345** [comment]
> Less work that atmos can do, therefor less reason to play the role if your not planning on doing autism projects in atmos.

when the fuck does atmos repair breaches anyways, its always engineering

> these sprites are really gross

codersprites

**Comment by maxion12345** [comment]
anyways, these bots aren't meant to replace the job of atmos techs. They don't fix breaches, they can't undo plasmafires, and you need a bunch of them to handle any breaches bigger than a 1x1. Just a neat robot that works together with other robots to accomplish a bigger task

**Comment by Cartlord** [comment]
They can kind-of undo a fire on their own though, just takes a while. They scrub the bad fire gas and stabilize temperature.

**Review by nesisknesis — APPROVED** [review]
yea

also original author's name here so they get credited in the changelog
image

**Comment by MegaEmpirical** [comment]
> In addition, emags are contraband and technically weapons under the new space law, so if security hands one over to cargo then admins gonna get salty.

Emags are tools, not wewpons.

**Comment by maxion12345** [comment]
> > In addition, emags are contraband and technically weapons under the new space law, so if security hands one over to cargo then admins gonna get salty.
> 
> Emags are tools, not wewpons.

aha, but you can emag a preternis to disable them, thusly making them a weapon under the new space law

**Comment by MegaEmpirical** [comment]
> > > In addition, emags are contraband and technically weapons under the new space law, so if security hands one over to cargo then admins gonna get salty.
> > 
> > 
> > Emags are tools, not wewpons.
> 
> aha, but you can emag a preternis to disable them, thusly making them a weapon under the new space law

I am telling you they're tools.

**Comment by maxion12345** [comment]
> > > > In addition, emags are contraband and technically weapons under the new space law, so if security hands one over to cargo then admins gonna get salty.
> > > 
> > > 
> > > Emags are tools, not wewpons.
> > 
> > 
> > aha, but you can emag a preternis to disable them, thusly making them a weapon under the new space law
> 
> I am telling you they're tools.

**Comment by maxion12345** [comment]
ah fuck I hit the wrong button

**Comment by JamieD1** [comment]
Feels extremely overpowered @SomeguyManperson

File: code/datums/components/crafting/recipes.dm [hunk 1]
```diff
@@ -248,6 +248,16 @@
...
 	category = CAT_ROBOT
 
+/datum/crafting_recipe/Atmosbot
+	name = "Atmosbot"
+	result = /mob/living/simple_animal/bot/atmosbot
+	reqs = list(/obj/item/analyzer = 1,
+				/obj/item/bodypart/r_arm/robot = 1,
+				/obj/item/assembly/prox_sensor = 1,
+				/obj/item/tank/internals = 1)
+	time = 40
```
Comment (Bibby0110) [inline code/datums/components/crafting/recipes.dm]:
```suggestion
	time = 4 SECONDS
```
maybe it should be longer?

File: code/game/objects/effects/temporary_visuals/miscellaneous.dm [hunk 2]
```diff
@@ -476,3 +476,10 @@
 /obj/effect/temp_visual/dir_setting/space_wind/Initialize(mapload, set_dir, set_alpha = 255)
 	. = ..()
 	alpha = set_alpha
+
+/obj/effect/temp_visual/vent_wind
+	icon = 'icons/effects/atmospherics.dmi'
+	icon_state = "vent_wind"
+	layer = FLY_LAYER
+	duration = 4.8
```
Comment (Bibby0110) [inline code/game/objects/effects/temporary_visuals/miscellaneous.dm]:
```suggestion
	duration = 0.48 SECONDS
```

**Review by Bibby0110 — COMMENTED** [review]
Didn't look through the rest honk

**Comment by maxion12345** [comment]
I changed the two "time =" i found to the SECONDS format. Crafting time has been increased to six seconds to be in line with other "complex" bots (ED209, Securitron). For reference, the other bots are 4 seconds. If you guys think its too overpowered, we can change it just to seal breaches, seeing as with monstermos the vents can do most of the work refilling a room.

**Comment by maxion12345** [comment]
@JamieD1 this gonna get testmerged?

**Comment by maxion12345** [comment]
These bots are just a neat addition to the game, even if they don't really do much. I wonder if I can make some voice lines for them.

**Comment by maxion12345** [comment]
notes so far:
they work pretty well, and can be mass produced easily. 

thats when they actually can reach the breach, sometimes they just stop for no reason mid-movement and refuse to take another step. thinking maybe of making a smart metal grenade bot that is a single-use suicide bomber on wheels that is crafted with a scooter, proximity sensor, and a smart metal grenade, that just wheels itself to the breach then blows itself up to foam it

**Comment by JamieD1** [comment]
@maxion12345 can you make them cover a breach with barriers? right now they place one barrier then give up..

**Comment by maxion12345** [comment]
@JamieD1 
new suicide bot with codersprite is completed

**Comment by adamsong** [comment]
Check linter please

**Comment by maxion12345** [comment]
> Check linter please

linter is green now

**Comment by maxion12345** [comment]
after 2 months in development i hope its been worth the wait

**Comment by JamieD1** [comment]
Will testmerge and see if its not shit.

**Comment by Bibby0110** [comment]
I don’t think it needs to be called an ASSBot

**Comment by maxion12345** [comment]
> I don’t think it needs to be called an ASSBot

because its funny

**Comment by Partheo** [comment]
Sprite is really bad and I personally dislike this feature cause I think breaches should be dangerous and should stay dangerous

**Pull request merged**